### PR TITLE
[BD-6] OEP-18 compliance

### PR DESCRIPTION
**Pull request by ericfab179**
Add setup.py install_requirements definition.
Update travis file using new requirements.
Create requirements folder.
create pip_tools, base, test and travis requirements files.
Add Makefile.
Add makefile upgrade command.
Include requirements files generated using upgrade command.
Add openedx.yaml and include OEPs list.
Add venv to gitignore.

**Reviewers**
- [ ] @awais786 
- [ ] @morenol

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ericfab179! I've created [OSPR-4504](https://openedx.atlassian.net/browse/OSPR-4504) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: setup.py [hunk 1]
```diff
@@ -63,6 +86,7 @@ def package_data(pkg, root_list):
         ]
     },
     package_data=package_data("recommender", ["static", "translations"]),
+    install_requires=load_requirements('requirements/base.in'),
```
Comment (morenol) [inline setup.py:89]: It seems that this package didn't have dependencies before, so  I guess that we don't changes on this file

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,8 @@ dist: xenial # required for Python >= 3.7 (travis-ci/travis-ci#9069)
 python:
 - 3.5
 - 3.8
-install: pip install -r requirements.txt
```
Comment (morenol) [inline .travis.yml:6]: Remove the requirements.txt file

**Comment by ericfab179** [comment]
@awais786 We think this is ready to be merged.

**Comment by edx-webhook** [comment]
@ericfab179 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/RecommenderXBlock&entry.867055334=https://github.com/edx/RecommenderXBlock/pull/19&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-05-09+21:46&entry.1917517419=2020-05-20+09:29&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**